### ISSUE TITLE
implement way to iterate over all labels of an AxisCollection

### ISSUE THREAD

**Issue by gdementen** [larray-project/larray#632]
The real goal is to iterate on all axes of an array except some, but I think that excluding axes should be done via `AxisCollection.__sub__`.

This already works but is a bit too complex to my tastes (requires an extra import):

```python
>>> from itertools import product
>>> arr = ndtest((2, 3, 4))
>>> arr
a  b\c  c0  c1  c2  c3
a0   b0   0   1   2   3
a0   b1   4   5   6   7
a0   b2   8   9  10  11
a1   b0  12  13  14  15
a1   b1  16  17  18  19
a1   b2  20  21  22  23
>>> for index in product(*(arr.axes - 'b')):
….    print(arr[index])
b  b0  b1  b2
    0   4   8
b  b0  b1  b2
    1   5   9
b  b0  b1  b2
    2   6  10
b  b0  b1  b2
    3   7  11
b  b0  b1  b2
   12  16  20
b  b0  b1  b2
   13  17  21
b  b0  b1  b2
   14  18  22
b  b0  b1  b2
   15  19  23
```
We could improve this by having an iterlabels method on AxisCollection:

```python
>>> for index in (arr.axes - 'b').iterlabels():
….    print(arr[index])
```
@alixdamman do you have any other idea?

**Comment by alixdamman** [comment]
Or we implement our own `product` function which tests if argument is an AxisCollection. 
If it's an AxisCollection, we return an iterator over labels, otherwise the usual `product` is called.

**Comment by gdementen** [comment]
I think we should avoid overriding builtin/stdlib functions as much as possible. I did this for sum/min/max/all/any because I saw no other option to implement with_total but here we have other options.

Besides having the user use product will not work when we support MultiIndex/sparse arrays, hence I prefer a method on AxisCollection. I can only think of iterlabels but if you have another name in mind...
